ui/components/app/name/name-details/name-details.tsx

```diff
@@ -54,19 +52,30 @@ import {
 } from '../../../../store/actions';
 import { useCopyToClipboard } from '../../../../hooks/useCopyToClipboard';
 import { useName } from '../../../../hooks/useName';
-import { I18nContext } from '../../../../contexts/i18n';
 import { useDisplayName } from '../../../../hooks/useDisplayName';
+import { useI18nContext } from '../../../../hooks/useI18nContext';
 import { usePetnamesMetrics } from './metrics';
 
 const UPDATE_DELAY = 1000 * 2; // 2 Seconds
 
+// The following lines avoid i18n "unused messages" lint error.
+// t('nameProvider_ens');
+// t('nameProvider_etherscan');
+// t('nameProvider_lens');
+// t('nameProvider_token');
+const getProviderLabel = (sourceId: string) => `nameProvider_${sourceId}`;
```

Comment:
Can this be a standard function under the other isolated functions like `generateComboOptions`?